[expo-local-authentication][android][bare] Build fails, cannot find symbols FEATURE_FACE or FEATURE_IRIS

## 🐛 Bug Report

### Summary of Issue (just a few sentences)
I have been using expo-local-authentication version 9.0.0 with a bare react native workflow. Upgrading to 9.1.1 gave me the following error when building with Android. 

```
/my-project/node_modules/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java:93: error: cannot find symbol
    if (mPackageManager.hasSystemFeature(PackageManager.FEATURE_FACE)) {
                                                       ^
  symbol:   variable FEATURE_FACE
  location: class PackageManager
/my-project/node_modules/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java:96: error: cannot find symbol
    if (mPackageManager.hasSystemFeature(PackageManager.FEATURE_IRIS)) {
                                                       ^
  symbol:   variable FEATURE_IRIS
  location: class PackageManager
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':expo-local-authentication:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

 Expo CLI 3.21.9 environment info:
    System:
      OS: macOS 10.15.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
      Yarn: 1.22.4 - ~/.nvm/versions/node/v10.15.3/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6241897
      Xcode: 11.5/11E608c - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 3.21.9

Building on Pixel 3a XL emulator with Android API 29.

### Expected behavior 
Build should not fail.

--- a/packages/expo-local-authentication/CHANGELOG.md
+++ b/packages/expo-local-authentication/CHANGELOG.md
@@ -10,6 +10,7 @@
 
 - Fix crash when `NSFaceIDUsageDescription` is not provided and device fallback is disabled. ([#8595](https://github.com/expo/expo/pull/8595) by [@tsapeta](https://github.com/tsapeta))
 - Added missing biometric permission to Android. ([#8692](https://github.com/expo/expo/pull/8692) by [@byCedric](https://github.com/byCedric))
+- Use hardcoded system feature strings to support Android SDK 28. ([#9034](https://github.com/expo/expo/pull/9034) by [@bycedric](https://github.com/bycedric))
 
 ## 9.1.1 — 2020-05-29
 

--- a/packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java
+++ b/packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java
@@ -87,15 +87,23 @@ public class LocalAuthenticationModule extends ExportedModule {
       promise.resolve(results);
       return;
     }
-    if (mPackageManager.hasSystemFeature(PackageManager.FEATURE_FINGERPRINT)) {
-      results.add(AUTHENTICATION_TYPE_FINGERPRINT);
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      if (mPackageManager.hasSystemFeature("android.hardware.fingerprint")) {
+        results.add(AUTHENTICATION_TYPE_FINGERPRINT);
+      }
     }
-    if (mPackageManager.hasSystemFeature(PackageManager.FEATURE_FACE)) {
-      results.add(AUTHENTICATION_TYPE_FACIAL_RECOGNITION);
+
+    if (Build.VERSION.SDK_INT >= 29) {
+      if (mPackageManager.hasSystemFeature("android.hardware.biometrics.face")) {
+        results.add(AUTHENTICATION_TYPE_FACIAL_RECOGNITION);
+      }
+
+      if (mPackageManager.hasSystemFeature("android.hardware.biometrics.iris")) {
+        results.add(AUTHENTICATION_TYPE_IRIS);
+      }
     }
-    if (mPackageManager.hasSystemFeature(PackageManager.FEATURE_IRIS)) {
-      results.add(AUTHENTICATION_TYPE_IRIS);
-    }
+
     promise.resolve(results);
   }
 

--- a/packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java
+++ b/packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java
@@ -88,6 +88,9 @@ public class LocalAuthenticationModule extends ExportedModule {
       return;
     }
 
+    // note(cedric): replace hardcoded system feature strings with constants from
+    // PackageManager when dropping support for Android SDK 28
+
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
       if (mPackageManager.hasSystemFeature("android.hardware.fingerprint")) {
         results.add(AUTHENTICATION_TYPE_FINGERPRINT);

